apps/internal/base/base.go

```diff
@@ -473,31 +482,9 @@ func (b Client) AuthResultFromToken(ctx context.Context, authParams authority.Au
 // was created to test the function against refreshin
 var GetCurrentTime = time.Now
 
-func (c Client) doesTenantExists(tenant string) bool {
-	_, exists := c.canRefresh[tenant]
-	return exists
-}
-
-func (c *Client) addTenantIntoCanRefresh(tenant string) {
-	c.canRefresh[tenant] = struct{}{}
-}
-
-func (c *Client) removeTenantFromCanRefresh(tenant string) {
-	delete(c.canRefresh, tenant)
-}
-
 // shouldRefresh returns true if the token should be refreshed.
-func (b Client) shouldRefresh(t time.Time, tId string) bool {
-	b.refreshMu.RLock()
-	if !b.doesTenantExists(tId) {
-		b.refreshMu.RUnlock()
-		b.refreshMu.Lock()
-		defer b.refreshMu.Unlock()
-		b.addTenantIntoCanRefresh(tId)
-		return !t.IsZero() && t.Before(GetCurrentTime())
-	}
-	b.refreshMu.RUnlock()
-	return false
+func (b *Client) shouldRefresh(t time.Time) bool {
```

Comment:
I'd inline this because it's trivial. If you still don't want to inline it, consider making it a plain old function because it doesn't need to reference `Client`

Reply:
Updated